### PR TITLE
improve logging fixed #580

### LINKED ISSUE #580: Improve logging to track run source (commit/config)
->  we should print config file path, which is started as log.info 
->  we see in log file on server which commit is used

### PR DESCRIPTION
**Pull request by zishanbilal**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/581)

### CODE REVIEW

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 1]
```diff
@@ -242,7 +243,7 @@ trait BeamHelper extends LazyLogging {
       case _ =>
         val (_, outputDirectory) = runBeamWithConfig(config)
         val props = new Properties()
-        props.setProperty("commitHash", LoggingUtil.getCommitHash)
+        props.setProperty("commitHash", getCommitHash)
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/BeamHelper.scala]: This is a bit of personal preference, but I would rather have this be `BashUtil.getCommitHash` as it makes it easier to understand that this logic is being provided elsewhere.

File: src/main/java/beam/utils/BashUtil.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package beam.utils;
```
Comment (JustinPihony) [inline src/main/java/beam/utils/BashUtil.java]: Any reason to make this a java file, and not scala?

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 3]
```diff
@@ -337,10 +338,16 @@ trait BeamHelper extends LazyLogging {
       beamConfig.beam.agentsim.simulationName,
       beamConfig.beam.outputs.addTimestampToOutputDirectory
     )
+
     LoggingUtil.createFileLogger(outputDirectory)
     matsimConfig.controler.setOutputDirectory(outputDirectory)
     matsimConfig.controler().setWritePlansInterval(beamConfig.beam.outputs.writePlansInterval)
 
+    logger.info(s"Starting beam on branch $getBranch at commit $getCommitHash.")
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/BeamHelper.scala]: While this is only done once, to remain consistent with the newer, more performant standard could you use `{}`:

`logger.info("Starting beam on branch {} at commit {}.", getBranch, getCommitHash)`

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 4]
```diff
@@ -337,10 +338,16 @@ trait BeamHelper extends LazyLogging {
...
+    val outConf = Paths.get(outputDirectory, "beam.conf")
+    Files.write(outConf, config.root().render(ConfigRenderOptions.concise()).getBytes)
+    logger.info(s"Config [${beamConfig.beam.agentsim.simulationName}] copied to $outConf.")
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/BeamHelper.scala]: What is the point of this? Just makes it easier to debug what was actually configured for that run?

Comment (zishanbilal) [reply src/main/scala/beam/sim/BeamHelper.scala]: to log them in output that is published to s3 for future analyses.

**Pull request merged**